control-plane-operator/hostedclusterconfigoperator/controllers/resources/recovery/recovery.go

```diff
@@ -0,0 +1,52 @@
+package recovery
+
+import (
+	"context"
+	"fmt"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func RecoverMonitoringStack(ctx context.Context, hcp *hyperv1.HostedControlPlane, c client.Client) error {
+	log := ctrl.LoggerFrom(ctx)
+	monitoringStackNS := "openshift-monitoring"
+	stsName := "prometheus-k8s"
+
+	// Check deployment of monitoring stack
+	prometheusSts := &appsv1.StatefulSet{}
+	err := c.Get(ctx, types.NamespacedName{Namespace: monitoringStackNS, Name: stsName}, prometheusSts)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get prometheus statefulSet %s in namespace %s: %w", stsName, monitoringStackNS, err)
+		}
+
+		return fmt.Errorf("prometheus statefulSet is still starting, rescheduling reconciliation: %w", err)
+	}
+
+	if prometheusSts.Status.ReadyReplicas < prometheusSts.Status.Replicas {
```

Comment:
I asked Juanma offline about the necessity of force deleting the pods when the ready replicas are still less than the replicas. He stated the pods got stuck after restoration because of the volumes are still associated with the AWS volumes which potentially will not exists. This blocks the monitoring CO stuck on deployment and not reporting status. Once we delete the pods, that association is removed and the CO could continue the deployment.